scalableminds/webknossos
# After merge, remove segment item of id that doesn't exist anymore

- When merging two segments, removes the segment from the segment list.
- The active segment/item now always survives the merge operation (wasn't the case when merging agglomerate-skeleton or skeletons with a shortcut).

### URL of deployed dev instance (used for testing):
- https://removesegmentaftermerge.webknossos.xyz

### Steps to test:
Part 1:
- create a new annotation for a dataset that has an agglomerate mapping (e.g., test-agglomerate)
- activate proofreading mode
- click on two segments that should be merged so that they have an entry each in the segments tab
- merge them with the context menu
- the segment id that doesn't exist anymore in the data should also not exist anymore in the segments tab
- repeat the above by loading agglomerate skeletons and merging them (make one node active and rightclick another node -> create edge)
Part 2:
- go into skeleton mode
- create two trees
- merge them with the ctrl+shift+alt shortcut --> active tree should survive
- merge them with the context menu --> active tree should survive

### Issues:
- fixes #7315

------
(Please delete unneeded items, merge only when none are left open)
